src/project/internal/projectactionscontroller.cpp

```diff
@@ -938,15 +918,145 @@ void ProjectActionsController::warnPublishIsNotAvailable()
+    buttons.push_back(saveCopyBtn);
+
+    int defaultBtn = saveCopyBtn.btn;
+
+    IInteractive::ButtonData revertToLastSavedBtn(saveCopyBtn.btn + 1, trc("project", "Revert to last saved"),
+                                                  true /*accent*/);
+
+    int btn = interactive()->error(title, body, errorText, buttons, defaultBtn).button();
+
+    if (btn == saveCopyBtn.btn) {
+        saveProjectCopy(true);
+    } else if (btn == revertToLastSavedBtn.btn) {
+        revertCorruptedScoreToLastSaved();
+    }
+}
+
+void ProjectActionsController::warnCorruptedScoreCannotBeSavedLocally(const SaveLocation& location, const std::string& errorText,
+                                                                      bool newlyCreated)
+{
+    std::string title = trc("project", "This score has become corrupted and contains errors");
+    std::string body = trc("project",
+                           "You can continue saving it locally, although the file may become unusable. "
+                           "To preserve your score, revert to the last saved version, or fix the errors manually. "
+                           "You can also get help for this issue on musescore.org.");
+
+    IInteractive::ButtonDatas buttons;
+    buttons.push_back(interactive()->buttonData(IInteractive::Button::Cancel));
+
+    IInteractive::ButtonData saveAnywayBtn(IInteractive::Button::CustomButton, trc("project", "Save anyway"), newlyCreated /*accent*/);
+    buttons.push_back(saveAnywayBtn);
+
+    int defaultBtn = saveAnywayBtn.btn;
+
+    IInteractive::ButtonData revertToLastSavedBtn(saveAnywayBtn.btn + 1, trc("project", "Revert to last saved"),
+                                                  true /*accent*/);
+    if (!newlyCreated) {
+        buttons.push_back(revertToLastSavedBtn);
+        defaultBtn = revertToLastSavedBtn.btn;
+    }
+
+    int btn = interactive()->error(title, body, errorText, buttons, defaultBtn).button();
+
+    if (btn == saveAnywayBtn.btn) {
+        saveProjectAt(location, SaveMode::Save, true);
+    } else if (btn == revertToLastSavedBtn.btn) {
+        revertCorruptedScoreToLastSaved();
+    }
+}
+
+void ProjectActionsController::warnCorruptedScoreUponOpenningCannotBeSaved(const std::string& errorText)
+{
+    std::string title = trc("project", "Your score cannot be saved");
+    std::string body = trc("project", "This score is corrupted. You can get help for this issue on musescore.org.");
+
+    IInteractive::ButtonData getHelpBtn(IInteractive::Button::CustomButton, trc("project", "Get help"), true /*accent*/);
+
+    int btn = interactive()->error(title, body, errorText, {
+        interactive()->buttonData(IInteractive::Button::Cancel),
+        getHelpBtn
+    }, getHelpBtn.btn).button();
+
+    if (btn == getHelpBtn.btn) {
+        interactive()->openUrl(configuration()->supportForumUrl());
+    }
+}
+
+void ProjectActionsController::revertCorruptedScoreToLastSaved()
```

Comment:
I'm missing this dialog from the design: 
<img width="499" alt="Scherm­afbeelding 2023-02-01 om 13 24 58" src="https://user-images.githubusercontent.com/48658420/216041861-399dc1a0-f04f-445b-9379-abd09e686fb2.png">